src/main/java/it/pagopa/transactions/utils/LogMaskerPatternLayout.java

```diff
@@ -0,0 +1,47 @@
+package it.pagopa.transactions.utils;
+
+import ch.qos.logback.classic.PatternLayout;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class LogMaskerPatternLayout extends PatternLayout {
+    private Pattern patternString;
+    private List<String> maskPatterns = new ArrayList<>();
+
+    public void addMaskPattern(String maskPattern) {
+        maskPatterns.add(maskPattern);
+        patternString = Pattern.compile(maskPatterns.stream().collect(Collectors.joining("|")), Pattern.MULTILINE);
+    }
```

Comment:
Where will be this method be called at runtime to add the mask patterns?

Reply:
It is not directly called by any other point in the project, but it is necessary to make the logback configuration file working, when it add the mask patterns to the list. If you try to comment that method and start the application, the start will fail.